pennylane/gradients/parameter_shift.py

```diff
@@ -757,7 +757,9 @@ def _get_var_with_second_order(pdA2, f0, pdA):
             and the gradient recipe contains an unshifted term, this value is used,
             saving a quantum evaluation.
     """
-    return qml.math.array(pdA2 - 2 * f0 * pdA)
+    if any(isinstance(var, np.ndarray) for var in [pdA2, f0, pdA]):

```

Comment:
Very minor: the items in the list are more so terms required to recreate the partial derivative of the variance of an operator, rather than the variance itself.
```suggestion
    if any(isinstance(term, np.ndarray) for term in [pdA2, f0, pdA]):
```